src/App.vue

```diff
@@ -170,43 +208,69 @@ export default {
         const jwt = this.getJWT()
         const keycloakRoles = this.getKeycloakRoles(jwt)
         this.setKeycloakRoles(keycloakRoles)
+        // safety check
+        if (!this.businessId && !this.nrNumber) throw new Error('Missing Business ID or NR Number')
       } catch (error) {
         // eslint-disable-next-line no-console
         console.error(error)
-        this.accountAuthorizationDialog = true
+        if (this.businessId) this.businessAuthErrorDialog = true
+        if (this.nrNumber) this.nameRequestAuthErrorDialog = true
         return // do not execute remaining code
       }
 
       // check if current user is authorized
       this.getAuthorizations().then(data => {
         this.storeAuthorizations(data) // throws if no role
 
-        // good so far ... fetch the rest of the data
-        Promise.all([
-          this.getBusinessInfo(),
-          this.getEntityInfo(),
-          this.getTasks(),
-          this.getFilings(),
-          this.getAddresses(),
-          this.getDirectors()
-        ]).then(data => {
-          if (!data || data.length !== 6) throw new Error('Incomplete data')
-          this.storeBusinessInfo(data[0])
-          this.storeEntityInfo(data[1])
-          this.storeTasks(data[2])
-          this.storeFilings(data[3])
-          this.storeAddresses(data[4])
-          this.storeDirectors(data[5])
-          this.dataLoaded = true
-        }).catch(error => {
-          // eslint-disable-next-line no-console
-          console.error(error)
-          this.dashboardUnavailableDialog = true
-        })
+        // good so far ... fetch the business data
```

Comment:
I like the comment, like a little code-pep-talk😃

Reply:
Another gem for you:
`// just let signin page do its thing`